VideoLocker/src/main/java/org/edx/mobile/view/CourseDiscussionResponsesFragment.java

```diff
@@ -148,4 +122,110 @@ public void onClickAddComment(@NonNull DiscussionComment comment) {
     public void onClickViewComments(@NonNull DiscussionComment comment) {
         router.showCourseDiscussionComments(getActivity(), comment, discussionThread.isClosed());
     }
+
+    public static class DiscussionResponsesLoader implements
+            InfiniteScrollUtils.PageLoader<DiscussionComment> {
+        private InfiniteScrollUtils.PageLoadCallback<DiscussionComment> callback;
+
+        private Context context;
+        private GetCommentListTask getCommentListTask;
+        private GetCommentListTask getEndorsedListTask;
+
+        private int nextPage = 1;
+        private String threadId;
+        private boolean isQuestionTypeThread;
+
+        private ThreadComments unendorsedComments;
+        private boolean isEndorsedFetched = false;
+
+        public DiscussionResponsesLoader(@NonNull Context context,@NonNull String threadId,
+                                         boolean isQuestionTypeThread) {
+            this.context = context;
+            this.threadId = threadId;
+            this.isQuestionTypeThread = isQuestionTypeThread;
+        }
+
+        @Override
+        public void loadNextPage(@NonNull InfiniteScrollUtils.PageLoadCallback<DiscussionComment> callback) {
+            this.callback = callback;
+            if (isQuestionTypeThread && !isEndorsedFetched) {
+                getEndorsedList();
+            }
+            getCommentList();
+
+        }
+
+        /**
+         * Gets the list of endorsed answers for a {@link DiscussionThread.ThreadType#QUESTION}
+         * type discussion thread
+         */
+        protected void getEndorsedList() {
+            if (getEndorsedListTask != null) {
+                getEndorsedListTask.cancel(true);
+            }
+            getEndorsedListTask = new GetCommentListTask(context, threadId, 1, isQuestionTypeThread,
+                    true) {
+                @Override
+                public void onSuccess(ThreadComments threadComments) {
+                    isEndorsedFetched = true;
+                    callback.onPartialPageLoaded(threadComments.getResults());
```

Comment:
As I [suggested][loader suggestion] previously, my vote would be to use the Loader framework. It seems perfectly suited for our purposes. Note that the Gist I linked there was just a proof of concept implementation, and I didn't actually test it yet. However, the general approach should work fine, and we only have to create a good `Loader` implementation once.

You mentioned that there are some issues with the Loader framework, but didn't elaborate there. If there are actually some real issues that can't be gotten around, I would be interested in learning about them. If you want to have a discussion on this, we can discuss it in the comments  of the [Code Improvements][] document.

[loader suggestion]: https://github.com/edx/edx-app-android/pull/372/files#r40550623
[Code Improvements]: https://openedx.atlassian.net/wiki/display/MA/Android+Code+Improvements